### PR TITLE
JITM: Add external link icon to CTAs that opens new tab.

### LINKED ISSUE #29552: JITM: CTAs that open new tabs need external link icons
### What's happening?

Certain JITM CTAs open new tabs, those that do, should have an external link icon:

Image

Image

### PR DESCRIPTION
**Pull request by jkguidaven**
Certain JITM CTAs open new tabs, those that do, should have an external link icon. This PR updates the JITM to render external link icon.

**Before**:
Screen Shot 2023-03-23 at 8 17 04 PM

**After**:
Screen Shot 2023-03-23 at 7 43 54 PM

Fixes #29552

## Proposed changes:
* Update JITM package to render external link icons when CTA opens a new tab.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
A P2 post discussing about how to improve the current JITM should be found in the link below:
p1HpG7-l9I-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

* Spin up a JN site and make sure you are running it with Jetpack Beta pointing to this branch.
* Connect your JN site to your WP account
* Once connected, go to JP dashboard screen (/wp-admin/admin.php?page=jetpack#/dashboard)
* Confirm JITM with 'Sign up' CTAs renders with external link icon. clicking this CTA should open a new tab.
Screen Shot 2023-03-23 at 9 03 35 PM
* Try to activate more Jetpack modules in My Jetpack and confirm that some of the CTAs also work as expected.
Screen Shot 2023-03-23 at 9 11 49 PM
* There are a few JITM with CTAs that do not open a new tab. The external link icon should not be rendered in this case. eg
Screen Shot 2023-03-23 at 9 11 49 PM

### CODE REVIEW

File: projects/packages/jitm/src/css/jetpack-admin-jitm.scss [hunk 1]
```diff
@@ -106,6 +106,12 @@ $jp-gray-20:             #a7aaad;
 		.gridicons-plus-small + .gridicon {
 			margin-left: -4px;
 		}
+
+		// Properly align icon with the button text
+		.gridicons-external-link {
+			margin: -3px 0 -3px 2px;
+			fill: #fff;
```
Comment (manzoorwanijk) [inline projects/packages/jitm/src/css/jetpack-admin-jitm.scss]:
```suggestion
			fill: $white;
```

**Review by manzoorwanijk — DISMISSED** [review]
Looks great

File: projects/packages/jitm/src/css/jetpack-admin-jitm.scss (hunk 1 above)
Comment (jkguidaven) [reply projects/packages/jitm/src/css/jetpack-admin-jitm.scss]: fixed. thanks 👍

File: projects/packages/jitm/src/js/jetpack-jitm.js [hunk 2]
```diff
@@ -90,6 +106,7 @@ jQuery( document ).ready( function ( $ ) {
 					( ajaxAction ? 'data-ajax-action="' + ajaxAction + '"' : '' ) +
 					'>' +
 					envelope.CTA.message +
+					(externalLink ? EXTERNAL_LINK_ICON : '')
```
Comment (jeherve) [inline projects/packages/jitm/src/js/jetpack-jitm.js]: In line with the rest of the file, I think there should be spaces here?

```suggestion
					( externalLink ? EXTERNAL_LINK_ICON : '' )
```

File: projects/packages/jitm/src/js/jetpack-jitm.js [hunk 3]
```diff
@@ -69,6 +84,7 @@ jQuery( document ).ready( function ( $ ) {
 				}
 
 				var ajaxAction = envelope.CTA.ajax_action;
+				var externalLink = envelope.CTA.newWindow && !ajaxAction;
```
Comment (jeherve) [inline projects/packages/jitm/src/js/jetpack-jitm.js]:
```suggestion
				var externalLink = envelope.CTA.newWindow && ! ajaxAction;
```

**Review by jeherve — CHANGES_REQUESTED** [review]
this tested well for me in Chrome, Firefox, Edge, and Safari. I could not test the updates to the lists in JITMs, I don't know that we have JITMs currently active with lists.

I left 2 tiny nitpicks below regarding code style, but other than that I think we should be good to go!

**Review by jeherve — DISMISSED** [review]

**Comment by dhasilva** [comment]
This may need a rebase because of https://github.com/Automattic/jetpack/pull/29749, just to be sure.

**Comment by jkguidaven** [comment]
@jeherve, thanks for fixing the coding style.

I have fixed the conflict. Can I have another approval again so I can merge the PR?  🙏

**Review by jeherve — APPROVED** [review]

**Pull request merged**